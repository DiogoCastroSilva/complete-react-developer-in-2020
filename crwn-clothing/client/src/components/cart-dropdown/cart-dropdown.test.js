import React from 'react';
import { shallow } from 'enzyme';

import CartDropdown from './cart-dropdown.component';
import CartItem from '../cart-item/cart-item.component';

import {
    CartDropdownButtonContainer,
    EmptyMessageContainer
} from './cart-dropdown.styles';


describe('testing CartDropdown component', () => {
    let wrapper;
    let mockHandledClick;
    const cartItems = [
        {
            imageUrl: 'https://store.storeimages.cdn-apple.com/4668/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;op_usm=0.5,0.5&amp;.v=1604021663000',
            price: '1179â‚¬',
            name: 'Iphone 12 Pro',
            quantity: 1
        }
    ];

    beforeEach(() => {
        mockHandledClick = jest.fn();
        
        const mockProps = {
            cartItems: cartItems,
            handleClick: mockHandledClick
        };

        wrapper = shallow(<CartDropdown {...mockProps} />);
    });

    it('should render CartDropdown component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should handle click event', () => {
        wrapper.find(CartDropdownButtonContainer).simulate('click');
        expect(mockHandledClick).toHaveBeenCalled();
    });

    it('should render the right number of cart items', () => {
        expect(wrapper.find(CartItem).length).toEqual(cartItems.length);
    });

    it('should render empty message', () => {
        const newMockProps = {
            cartItems: [],
            handleClick: mockHandledClick
        };

        const newWrapper = shallow(<CartDropdown {...newMockProps} />);

        expect(newWrapper.exists(EmptyMessageContainer)).toBe(true);
    });

});

