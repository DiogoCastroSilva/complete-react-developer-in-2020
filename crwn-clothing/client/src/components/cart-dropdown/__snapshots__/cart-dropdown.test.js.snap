// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CartDropdown component should render CartDropdown component 1`] = `
<styled.div>
  <styled.div>
    <Memo(CartItem)
      item={
        Object {
          "imageUrl": "https://store.storeimages.cdn-apple.com/4668/as-images.apple.com/is/iphone-12-pro-family-hero?wid=940&amp;hei=1112&amp;fmt=jpeg&amp;qlt=80&amp;op_usm=0.5,0.5&amp;.v=1604021663000",
          "name": "Iphone 12 Pro",
          "price": "1179â‚¬",
          "quantity": 1,
        }
      }
    />
  </styled.div>
  <Styled(CustomButton)
    onClick={[MockFunction]}
  >
    GO TO CHECKOUT
  </Styled(CustomButton)>
</styled.div>
`;
